app/src/main/java/com/simply/birthdayapp/presentation/MainActivity.kt

```diff
@@ -3,14 +3,48 @@ package com.simply.birthdayapp.presentation
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.simply.birthdayapp.data.NetworkMonitor
 import com.simply.birthdayapp.presentation.navigation.RootNavigation
+import com.simply.birthdayapp.presentation.ui.components.NoNetworkConnectionDialog
 import com.simply.birthdayapp.presentation.ui.theme.BirthdayAppTheme
+import kotlinx.coroutines.delay
+import org.koin.android.ext.android.inject
 
 class MainActivity : ComponentActivity() {
+    private val networkMonitor: NetworkMonitor by inject()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
             BirthdayAppTheme {
+                val connectedToNetwork by networkMonitor.connected.collectAsStateWithLifecycle(
+                    initialValue = networkMonitor.isCurrentlyConnected(),
+                    lifecycleOwner = this,
+                )
+                var keepShowingNetworkDialog by rememberSaveable { mutableStateOf(true) }
+                var showNetworkDialog by rememberSaveable { mutableStateOf(true) }
```

Comment:
Why are the default values set to 'true'?

Reply:
By default, a dialog appears when the user is disconnected.

The dialog can be `dismissed` and `shown again` after 10 seconds.
The dialog `will not be shown` again until the app is restarted if the user clicks `Don't show`.

That is why these default values are set to `true`.

P.S. Have renamed that variables to align them with the user actions `Do not show`, `Dismiss`.